lib/chef-dk/policyfile/local_lock_fetcher.rb

```diff
@@ -75,6 +75,7 @@ def apply_locked_source_options(options_from_lock)
       # @return [String] of the policyfile lock data
       def lock_data
         FFI_Yajl::Parser.new.parse(content).tap do |data|
+          validate_revision_id(data["revision_id"])
           data["cookbook_locks"].each do |cookbook_name, cookbook_lock|
```

Comment:
I wonder if it would be worth putting `FFI_Yajl::Parser.new.parse(content)` into a `fetch_lock_data` method then we could have a single implementation of `lock_data` here instead of duplicating it in both local and remote? Not a huge deal and I don't think I'd block this pr for it but given the logic and validation is non-trivial it might be nice to keep it in one place.

Reply:
I've refactored the code and updated the PR with your suggestion.